ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/common/fragments/BaseListFragmentTest.java

```diff
@@ -0,0 +1,782 @@
+        //Arrange
+        doReturn(false).when(mockCustomStateViewHelper).hasLoadingView();
+        doNothing().when(mockDefaultStateViewHelper).showLoadingView();
+        doNothing().when(mockDefaultStateViewHelper).hideLoadingView();
+        doNothing().when(mockDefaultStateViewHelper).hideEmptyView();
+        PowerMockito.suppress(methods(BaseListFragment.class, "hideListView"));
+        doNothing().when(mockListPageChangeStateListener).onListPageStateChanged(ListPageState.LOADING);
+        //Act
+        testBaseListFragment.showLoadingViewAndHideOthers();
+        //Assert
+        Assert.assertTrue(isMethodSuccessfullyRun);
+    }
+
+    @Test
+    public void showErrorViewAndHideOthersTestShowErrorView() {
+        //Arrange
+        doReturn(true).when(mockCustomStateViewHelper).hasErrorView();
+        doNothing().when(mockCustomStateViewHelper).showErrorView();
+        doNothing().when(mockDefaultStateViewHelper).hideLoadingView();
+        doNothing().when(mockDefaultStateViewHelper).hideEmptyView();
+        doNothing().when(mockDefaultStateViewHelper).hideLoadingView();
+        doNothing().when(mockDefaultStateViewHelper).hideEmptyView();
+        PowerMockito.suppress(methods(BaseListFragment.class, "hideListView"));
+        doNothing().when(mockListPageChangeStateListener).onListPageStateChanged(ListPageState.ERROR);
+        //Act
+        testBaseListFragment.showErrorViewAndHideOthers();
+        //Assert
+        Assert.assertTrue(isMethodSuccessfullyRun);
+    }
+
+    @Test
+    public void showErrorViewAndHideOthersTestHideErrorView() {
+        //Arrange
+        doReturn(false).when(mockCustomStateViewHelper).hasErrorView();
+        doNothing().when(mockDefaultStateViewHelper).hideLoadingView();
+        doNothing().when(mockDefaultStateViewHelper).hideEmptyView();
+        doNothing().when(mockDefaultStateViewHelper).hideLoadingView();
+        doNothing().when(mockDefaultStateViewHelper).hideEmptyView();
+        PowerMockito.suppress(methods(BaseListFragment.class, "hideListView"));
+        doNothing().when(mockListPageChangeStateListener).onListPageStateChanged(ListPageState.ERROR);
+        //Act
+        testBaseListFragment.showErrorViewAndHideOthers();
+        //Assert
+        Assert.assertTrue(isMethodSuccessfullyRun);
+    }
+
+    @Test
+    public void showListViewAndHideOthersTest() {
+        //Arrange
+        PowerMockito.suppress(methods(BaseListFragment.class, "showListView"));
+        doNothing().when(mockDefaultStateViewHelper).hideEmptyView();
+        doNothing().when(mockDefaultStateViewHelper).hideLoadingView();
+        doNothing().when(mockDefaultStateViewHelper).hideEmptyView();
+        doNothing().when(mockListPageChangeStateListener).onListPageStateChanged(ListPageState.LIST);
+        //Act
+        testBaseListFragment.showListViewAndHideOthers();
+        //Assert
+        Assert.assertTrue(isMethodSuccessfullyRun);
```

Comment:
⚠️  It could be a good idea to import the `static` assertions. This way the code becomes smaller and avoids repetition.  

**[More details](https://confluence.devfactory.com/x/MxNrEw)**
          
## The possible solution in this case

```JAVA
import static org.junit.Assert.assertTrue;
...
assertTrue(isMethodSuccessfullyRun); 
```
 

<a id="06fcbf00-50da-11e8-89e2-ed64d7ac7be5" href="none" codereview-uuid-reserved-tag />

Reply:
done